File: examples/node/search_example.js
/**
 * Search restaurants in Madrid via the POS.DO MCP server (JSON-RPC over HTTP).
 *
 * Run: node search_example.js
 *
 * No dependencies — uses Node 18+ global fetch.
 */

const ENDPOINT = "https://pos.do/mcp";

async function callTool(name, args) {
  const response = await fetch(ENDPOINT, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      jsonrpc: "2.0",
      id: 1,
      method: "tools/call",
      params: { name, arguments: args },
    }),
  });

  if (!response.ok) {
    throw new Error(`HTTP ${response.status}`);
  }

  const rpc = await response.json();
  if (rpc.error) {
    throw new Error(`MCP error: ${JSON.stringify(rpc.error)}`);
  }
  // tools/call wraps the JSON payload as a string inside content[0].text.
  return JSON.parse(rpc.result.content[0].text);
}

(async () => {
  const result = await callTool("search_restaurants", {
    city: "Madrid",
    cuisine: "italian",
    features: "terrace",
    limit: 5,
  });
  console.log(`Found ${result.count} restaurants near ${result.city}:\n`);
  for (const r of result.restaurants || []) {
    console.log(
      `  ${r.name.padEnd(40)}  rating=${r.google_rating}  cuisine=${r.cuisine_type}`
    );
  }
})();

File: examples/python/search_example.py
"""Search restaurants in Madrid via the POS.DO MCP server (JSON-RPC over HTTP).

Run: python search_example.py

Requires: requests (`pip install requests`).
"""

import json
import sys

import requests

ENDPOINT = "https://pos.do/mcp"


def call_tool(name: str, arguments: dict) -> dict:
    """Invoke an MCP tool via tools/call. Returns the parsed result payload."""
    response = requests.post(
        ENDPOINT,
        headers={"Content-Type": "application/json"},
        json={
            "jsonrpc": "2.0",
            "id": 1,
            "method": "tools/call",
            "params": {"name": name, "arguments": arguments},
        },
        timeout=15,
    )
    response.raise_for_status()
    rpc = response.json()
    if "error" in rpc:
        raise RuntimeError(f"MCP error: {rpc['error']}")
    # tools/call wraps the JSON payload as a string inside content[0].text.
    text = rpc["result"]["content"][0]["text"]
    return json.loads(text)


def main() -> int:
    result = call_tool(
        "search_restaurants",
        {
            "city": "Madrid",
            "cuisine": "italian",
            "features": "terrace",
            "limit": 5,
        },
    )
    print(f"Found {result.get('count', 0)} restaurants near {result['city']}:\n")
    for r in result.get("restaurants", []):
        print(
            f"  {r.get('name'):<40}  "
            f"rating={r.get('google_rating')}  "
            f"cuisine={r.get('cuisine_type')}"
        )
    return 0


if __name__ == "__main__":
    sys.exit(main())
